base_classes/NXobject.nxdl.xml

```diff
@@ -27,8 +27,43 @@
 	name="NXobject" 
 	type="group">
 	<doc>
-		This is the base object of NeXus
+        This is the base object of NeXus. The groups and fields contained 
+        within this file are allowed to be present in any derived base class.
+
+        If nameType="partial", the placeholders (e.g., FIELDNAME or GROUPNAME)
+        can be replaced by the name of any object (field or group, 
+        respectively) that exists within the same group.
 	</doc>
-        <!--attribute name="name"><doc>name of instance</doc></attribute-->
+    <group type="NXcollection" minOccurs="0" />
+    <group type="NXdata" minOccurs="0" />
+    <group type="NXgeometry" minOccurs="0" />
+    <group type="NXlog" minOccurs="0" />
+    <group type="NXnote" minOccurs="0" />
+    <group type="NXparameters" minOccurs="0" />
+    <group type="NXtransformations" minOccurs="0" />
```

Comment:
Do we also need a `depends_on` attribute?

Reply:
I'm not sure why we would need to add it, since attributes are usually optional. Even the NXtransformations class appears in many application definitions without one.